cassandra-manage.py

```diff
@@ -0,0 +1,219 @@
+
+    def delete(self):
+        # Original method ignores 404 code for some reason, but we don't.
+        r = self.api.delete(**self.api_kwargs())
+        self.api.raise_for_status(r)
+
+    def validate(self):
+        LOG.debug('Check that object exists on server.')
+        self.exists(ensure=True)
+
+
+class CassandraDaemonSet(K8SResource, DaemonSet):
+    pods_required = 3
+    nodes_required = 3
+
+    def _require_nodes(self, expected):
+        nodes = self._get_nodes()
+        actual = len(nodes)
+        if actual < expected:
+            raise KubernetesError(
+                'Expected {} nodes with "{}" labels. '
+                'Actual {}.'.format(expected,
+                                    self._labels_to_string(),
+                                    actual))
+        return nodes
+
+    def _require_pods(self, expected):
+        pods = self._get_pods()
+        active_pods = [p for p in pods if operator.attrgetter('ready')]
+        actual = len(active_pods)
+        if actual < expected:
+            raise KubernetesError(
+                'Expected {} active pods with "{}" labels. '
+                'Actual {}.'.format(expected,
+                                    self._labels_to_string(),
+                                    actual))
+        return pods
+
+    def _get_nodes(self):
+        return Node.objects(self.api).filter(selector=self.labels)
+
+    def _get_pods(self):
+        return Pod.objects(self.api).filter(selector=self.labels)
+
+    def create(self):
+        nodes = self._require_nodes(self.nodes_required)
+        for ct in self.obj['spec']['template']['spec']['containers']:
+            if ct['name'] == 'cassandra':
+                ct['env'].append({'name': 'CASSANDRA_SEEDS',
+                                  'value': ','.join([n.name for n in nodes])})
+                break
+        super().create()
+
+    def delete(self):
+        super(CassandraDaemonSet, self).delete()
+        pods = self._get_pods()
+        LOG.debug('Cascade delete {} pods '
+                  'labeled with "{}'.format(len(pods),
+                                            self._labels_to_string()))
+        for pod in pods:
+            pod.delete()
+
+    def validate(self):
+        super().validate()
+
+        LOG.debug('Check that at least {} '
+                  'nodes exists.'.format(self.nodes_required))
+        self._require_nodes(self.nodes_required)
+
+        LOG.debug('Check that at least {} '
+                  'pods exists.'.format(self.pods_required))
+        self._require_pods(self.pods_required)
+
+
+class CassandraService(K8SResource, Service):
+    pass
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description='Yagoda cassandra '
+                                                 'management tool.')
+    parser.add_argument('action', choices=['create', 'update',
+                                           'delete', 'validate'])
+    group = parser.add_mutually_exclusive_group()
```

Comment:
`group = parser.add_mutually_exclusive_group(required=True)`, because there's no default action in main.